warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryFieldsVisitor.java

```diff
@@ -156,7 +157,12 @@ public Object visit(ASTIdentifier node, Object data) {
     @Override
     public Object visit(ASTFunctionNode node, Object data) {
         JexlArgumentDescriptor desc = JexlFunctionArgumentDescriptorFactory.F.getArgumentDescriptor(node);
-        Set<String> fields = desc.fields(helper, null);
+        Set<String> fields = null;
+        try {
+            fields = desc.fields(helper, null);
+        } catch (TableNotFoundException | InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
         ((Set<String>) data).addAll(fields);
```

Comment:
if fields is null, this is going to throw a NPE.  maybe move addAll into the try block?

Reply:
Another oversight - good catch.